### ISSUE TITLE
Unable to receive DHCP address from Unifi Wifi

### ISSUE THREAD

**Issue by delorean02** [CraftComputing/ProjectAxeEffect#14]
My home Wifi is a Unifi system.  

I just recieved my shipment with two devices.  The first one I tried would not receive a DHCP address from my home Wifi, nor would it receive one from a mobile router Wifi.  
I tried upgrading the FW to the version 3 posted in the GitHub, and still no joy.

I took out the second unit, did not upgrade the FW, and was still not able to receive a DHCP address from my home Wifi or travel router Wifi.  I was able to get it to receive a DHCP address from my cell phone hotspot.

My networks all have a special charater in the SSID. I setup a new network without the special character, and that still did not resolve the issue.  I've verified with exisitng devices that I can remove and add them back to Wifi, and they pull a DHCP address with no issue.  

So I tried using a static IP address.  I set it to a static IP, wrote the config, and when I checked the Info page, the IP was still 0.0.0.0.  But now also, the SNMP port changed to 0, and the SNMP community became blank.

I tried resetting the device, and still the IP remained 0.0.0.0, the SNMP port remained 0, and the SNMP community remained blank.  I then tried re-configuing with the same settings, and while the SNMP community did change, the IP address did not.

I then tried doing a factory reset, re-entered all the settings, and they appeared to finally take.  However, the SNMP port was still set to 0.  Although it does now appear on the network, and I can ping the device, I cannot read SNMP from it.

Any help here would be greatly appreciated.

**Comment by delorean02** [comment]
Logs from the device
000182691 [INFO] Connected
000182691 [INFO] IP Address: 0.0.0.0
000182861 [INFO] Connecting to Wi-Fi network REDACTED
000182868 [INFO] Waiting 30 seconds before retrying
000212877 [INFO] Connected
000212877 [INFO] IP Address: 0.0.0.0
000213047 [INFO] Connecting to Wi-Fi network REDACTED
000213054 [INFO] Waiting 30 seconds before retrying
000243057 [INFO] Connected
000243058 [INFO] IP Address: 0.0.0.0
000243228 [INFO] Connecting to Wi-Fi network REDACTED
000243234 [INFO] Waiting 30 seconds before retrying

**Comment by delorean02** [comment]
More logs after upgrading to FW 1.3
037447534 [INFO] Console deactivated
037451102 [DEBUG] Heartbeat
037451461 [DEBUG] Current temperature: 25.08 C
037451461 [DEBUG] Current humidity: 46.636 %RH
037456068 [DEBUG] if_status_callback: flags=0x3E
037456068 [INFO] Connected
037456068 [INFO] IP Address: 0.0.0.0
037456238 [INFO] Connecting to Wi-Fi network REDACTED
037456244 [INFO] Waiting 30 seconds before retrying
037456271 [DEBUG] Heartbeat
037456473 [DEBUG] Current temperature: 25.08 C
037456473 [DEBUG] Current humidity: 46.579 %RH
037459611 [DEBUG] if_link_callback: flags=0x3F
037459611 [DEBUG] if_link_callback: link is up
037461280 [DEBUG] Heartbeat
037461485 [DEBUG] Current temperature: 25.07 C
037461485 [DEBUG] Current humidity: 46.624 %RH
037466285 [DEBUG] Heartbeat
037466496 [DEBUG] Current temperature: 25.05 C
037466497 [DEBUG] Current humidity: 46.577 %RH
037471291 [DEBUG] Heartbeat
037471508 [DEBUG] Current temperature: 25.05 C
037471508 [DEBUG] Current humidity: 46.565 %RH
037476296 [DEBUG] Heartbeat
037476520 [DEBUG] Current temperature: 25.04 C
037476520 [DEBUG] Current humidity: 46.597 %RH
037481301 [DEBUG] Heartbeat
037481532 [DEBUG] Current temperature: 25.01 C
037481532 [DEBUG] Current humidity: 46.653 %RH
037486247 [DEBUG] if_status_callback: flags=0x3E
037486247 [INFO] Connected
037486247 [INFO] IP Address: 0.0.0.0
037486417 [INFO] Connecting to Wi-Fi network REDACTED
037486423 [INFO] Waiting 30 seconds before retrying

**Comment by delorean02** [comment]
Logs from the unit connecting successfully to phone hotspot.
000021603 [DEBUG] Heartbeat
000026105 [DEBUG] if_link_callback: flags=0x3F
000026105 [DEBUG] if_link_callback: link is up
000026165 [DEBUG] Current temperature: 24.98 C
000026165 [DEBUG] Current humidity: 47.594 %RH
000026608 [DEBUG] Heartbeat
000027110 [DEBUG] if_status_callback: flags=0x3F
000027110 [INFO] Connected
000027110 [INFO] IP Address: 192.168.124.91
000031177 [DEBUG] Current temperature: 24.98 C
000031177 [DEBUG] Current humidity: 47.525 %RH
000031639 [DEBUG] Heartbeat
000036189 [DEBUG] Current temperature: 25.00 C
000036189 [DEBUG] Current humidity: 47.481 %RH

**Comment by OpenTechNW** [comment]
I had the same issue with unifi... the workaround that worked for me (which i would have done anyway) is to set a DHCP reservation. 
find the device in the Client Devices list and go to the device settings and check 'Fixed IP Address' and give it an IP then reboot the AxeEffect

**Comment by delorean02** [comment]
I tried setting a DHCP reservation.  But still no joy. The issue appears to be the device connects to WiFi for a random amount of time, and then disconnects. I've even locked it down to a single AP.  I wish the logging on the device itself was a little more robust.

**Comment by delorean02** [comment]
SUCCESS!!!
OK, so I'm running a UDM Pro with software version 4.0.6 and Network version 8.3.32.  In the client settings there is an option above Fixed IP Address is Virtual Network Override.  I had to enable this, and manually choose the network the device should connect to.  This fixed the connectivity issue, and I had to restart it one more time to get the SNMP to start responding.  But I now have temperature and humidty reporting to Home Assistant!  Thanks for all the ideas and thoughts everyone!!
